src/main/java/com/google/devtools/build/lib/rules/python/PyBinary.java

```diff
@@ -140,7 +141,10 @@ private static Runfiles collectCommonRunfiles(RuleContext ruleContext, PyCommon
     }
     semantics.collectDefaultRunfiles(ruleContext, builder);
     builder.add(ruleContext, PythonRunfilesProvider.TO_RUNFILES);
-    builder.setEmptyFilesSupplier(PythonUtils.GET_INIT_PY_FILES);
+
+    if (ruleContext.attributes().get("legacy_create_init", Type.BOOLEAN)) {
+      builder.setEmptyFilesSupplier(PythonUtils.GET_INIT_PY_FILES);
+    }
```

Comment:
I'm a little dumbfounded that it's this simple.  I guess that means the Bazel code is well structured.

Reply:
Indeed it is, if I can do it then everyone can :)